src/System.Resources.Analyzers/Core/MarkAssembliesWithNeutralResourcesLanguage.cs

```diff
@@ -13,26 +16,124 @@ public abstract class MarkAssembliesWithNeutralResourcesLanguageAnalyzer : Diagn
     {
         internal const string RuleId = "CA1824";
 
+        protected const string GeneratedCodeAttribute = "GeneratedCodeAttribute";
+        protected const string StronglyTypedResourceBuilder = "StronglyTypedResourceBuilder";
+        private const string Designer = ".Designer.";
+
         private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(SystemResourcesAnalyzersResources.MarkAssembliesWithNeutralResourcesLanguageTitle), SystemResourcesAnalyzersResources.ResourceManager, typeof(SystemResourcesAnalyzersResources));
-        
+
         private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(SystemResourcesAnalyzersResources.MarkAssembliesWithNeutralResourcesLanguageMessage), SystemResourcesAnalyzersResources.ResourceManager, typeof(SystemResourcesAnalyzersResources));
         private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(SystemResourcesAnalyzersResources.MarkAssembliesWithNeutralResourcesLanguageDescription), SystemResourcesAnalyzersResources.ResourceManager, typeof(SystemResourcesAnalyzersResources));
-        
+
         internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Performance,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb385967.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
+        protected abstract void RegisterAttributeAnalyzer(CompilationStartAnalysisContext context, Action onResourceFound);
+
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
```

Comment:
You are setting this flag because this analyzer analyzes a generated file (Designer.cs) and you want to set it explicitly so that if the default changes or if there's an user option that changes the default, you still need to be called for the Designer file - correct? It's worth adding that comment. I was initially confused - I thought you were setting this flag to assert that you don't want to report diagnostics on generated files which is really not the case for this rule (the diagnostic either doesn't have a location or is on the attribute which we should report anyway).